src/battle_script_commands.c

```diff
@@ -5858,9 +5858,9 @@ static void Cmd_moveend(void)
                     u8 battler = battlers[i];
                     // Attacker is the damage-dealer, battler is mon to be switched out
                     if (IsBattlerAlive(battler)
+                      && gBattlerAttacker != battler
                       && GetBattlerHoldEffect(battler, TRUE) == HOLD_EFFECT_EJECT_BUTTON
-                      && !DoesSubstituteBlockMove(gBattlerAttacker, battler, gCurrentMove)
-                      && (gSpecialStatuses[battler].physicalDmg != 0 || gSpecialStatuses[battler].specialDmg != 0)
+                      && TARGET_TURN_DAMAGED
```

Comment:
is this correct? TARGET_TURN_DAMAGED will check gBattlerTarget and not the `batter` loop index

Reply:
oh, you are right. sorry :grimacing: